src/Npgsql/NpgsqlConnection.cs

```diff
@@ -617,14 +617,7 @@ public new ValueTask<NpgsqlTransaction> BeginTransactionAsync(IsolationLevel lev
             if (cancellationToken.IsCancellationRequested)
 		        return new ValueTask<NpgsqlTransaction>(Task.FromCanceled<NpgsqlTransaction>(cancellationToken));
 
-            try
-            {
-                return new ValueTask<NpgsqlTransaction>(BeginTransaction(level));
-            }
-            catch (Exception exception)
-            {
-                return new ValueTask<NpgsqlTransaction>(Task.FromException<NpgsqlTransaction>(exception));
-            }
+            return BeginTransaction(level, true, cancellationToken);
```

Comment:
Now that this is really async, you also need to add NoSynchronizationContextScope above as with other async surface APIs.